core/src/main/java/org/elasticsearch/action/admin/cluster/shards/TransportClusterSearchShardsAction.java

```diff
@@ -67,12 +76,20 @@ protected ClusterSearchShardsResponse newResponse() {
     }
 
     @Override
-    protected void masterOperation(final ClusterSearchShardsRequest request, final ClusterState state, final ActionListener<ClusterSearchShardsResponse> listener) {
+    protected void masterOperation(final ClusterSearchShardsRequest request, final ClusterState state,
+                                   final ActionListener<ClusterSearchShardsResponse> listener) {
         ClusterState clusterState = clusterService.state();
         String[] concreteIndices = indexNameExpressionResolver.concreteIndexNames(clusterState, request);
         Map<String, Set<String>> routingMap = indexNameExpressionResolver.resolveSearchRouting(state, request.routing(), request.indices());
+        Map<String, AliasFilter> indicesAndFilters = new HashMap<>();
+        for (String index : concreteIndices) {
+            AliasFilter aliasFilter = indicesService.buildAliasFilter(clusterState, index, request.indices());
```

Comment:
fair enough I was more concernd about some rendering overhead. I am fine with what we have